lightning/src/ln/channelmanager.rs

```diff
@@ -281,6 +285,7 @@ impl MsgHandleErrInternal {
 					},
 				},
 			},
+			chan_id: Some(channel_id),
```

Comment:
I think this could cause a `ChannelClosed` event to be generated [here](https://github.com/rust-bitcoin/rust-lightning/pull/997/commits/c32527902660a2e5e9b48baee40bc998e8c2c5a0#diff-645d12c7269d09caab57d2e0c1fd34f672ee259994c86ad4bad7f4da9183671cR783)? Which I don't think `from_chan_no_close` should do

Reply:
I moved the `ChannelClosed` event generation conditional on `shutdown_finish.is_some()` in `handle_error`. So I think it doesn't matter if this field is some/none anymore ? I let it as None(), if someone wants to re-use it for the future, they could tweak it, e.g log `from_chan_no_close`.